tests/envs/test_narrow_passage.py

```diff
@@ -0,0 +1,287 @@
+    env.render_state(traj.states[4], task, caption="caption")
+    # Render state at end of trajectory
+    env.render_state(traj.states[-1], task)
+
+
+def test_narrow_passage_collisions():
+    """Test that the robot can't go through walls or closed door."""
+    # set up environment
+    utils.reset_config({"env": "narrow_passage"})
+    env = NarrowPassageEnv()
+    door_type, _, robot_type, _, _ = sorted(env.types)
+
+    # Test robot should not be able to walk through each wall nor the door
+    sample_task = env.get_train_tasks()[0]
+    test_robot_xs = [0.1, 0.35, 0.7, 0.9]
+    y_midpoint = env.y_lb + (env.y_ub - env.y_lb) / 2
+    down_action = Action(np.array([0, -0.08, 0]).astype(np.float32))
+    for test_x in test_robot_xs:
+        state = sample_task.init.copy()
+        robot, = state.get_objects(robot_type)
+        state.set(robot, "x", test_x)
+        # try going down a lot
+        for _ in range(50):
+            state = env.simulate(state, down_action)
+        # make sure robot is still above the wall
+        robot_y = state.get(robot, "y")
+        assert robot_y > y_midpoint, f"Robot did not collide at x={test_x}"
+
+    # Test robot can walk through the doorway when the door is open
+    state = sample_task.init.copy()
+    robot, = state.get_objects(robot_type)
+    door, = state.get_objects(door_type)
+    state.set(robot, "x", 0.35)
+    state.set(door, "open", 1)  # set the door to be open
+    # try going down a lot
+    for _ in range(50):
+        state = env.simulate(state, down_action)
+    # make sure robot is below the wall now but not out of bounds
+    robot_y = state.get(robot, "y")
+    assert robot_y < y_midpoint, "Robot wasn't able to pass through doorway"
+    assert robot_y >= env.y_lb, "Robot went out of bounds"
+
+
+def test_narrow_passage_options():
+    """Test ParameterizedOptions work."""
+    # Set up environment
+    utils.reset_config({
+        "env": "narrow_passage",
+        "render_state_dpi": 150,
+    })
+    env = NarrowPassageEnv()
+    DoorIsOpen, TouchedGoal = sorted(env.predicates)
+    MoveAndOpenDoor, MoveToTarget = sorted(env.options)
+    door_type, _, robot_type, target_type, _ = sorted(env.types)
+
+    task = env.get_train_tasks()[0]
+    state = task.init
+    door, = state.get_objects(door_type)
+    robot, = state.get_objects(robot_type)
+    target, = state.get_objects(target_type)
+
+    # Test MoveAndOpenDoor then MoveToTarget
+    option_plan = [
+        MoveAndOpenDoor.ground([robot, door], [0.3]),
+        MoveToTarget.ground([robot, target], [0.5]),
+    ]
+    policy = utils.option_plan_to_policy(option_plan)
+    monitor = utils.SimulateVideoMonitor(task, env.render_state)
+    traj = utils.run_policy_with_simulator(
+        policy,
+        env.simulate,
+        task.init,
+        lambda _: False,
+        max_num_steps=1000,
+        exceptions_to_break_on={utils.OptionExecutionFailure},
+        monitor=monitor,
+    )
+    # Save video of run
```

Comment:
I assume this video creation was for debugging; you can comment it out for future use or just remove it